apps/meters_to_ha/meters_to_ha.py

```diff
@@ -1746,115 +1754,153 @@ def get_gazpar_file(self):
             # Wait for Connexion #####
             self.mylog("Connexion au site GRDF", end="")
 
+            self.mylog("Get url", end="")
             self.__browser.get(self.__class__.site_grdf_url)
             self.mylog(st="OK")
 
-            # Wait for Password #####
-            self.mylog("Waiting for Password", end="")
-
-            ep = EC.presence_of_element_located((By.ID, "pass"))
-            el_password = self.__wait.until(
-                ep,
-                message="failed, page timeout (timeout="
-                + str(self.configuration[PARAM_TIMEOUT])
-                + ")",
-            )
-            self.mylog(st="OK")
-
             # Wait for Email #####
             self.mylog("Waiting for Email", end="")
-            ep = EC.presence_of_element_located((By.ID, "mail"))
+            ep = EC.presence_of_element_located((By.ID, "input27"))
             el_email = self.__wait.until(
                 ep,
                 message="failed, page timeout (timeout="
                 + str(self.configuration[PARAM_TIMEOUT])
                 + ")",
             )
             self.mylog(st="OK")
-
+            
+            if self.configuration[PARAM_SCREENSHOT]:
+                self.get_screenshot("01_screenshot_before_user.png")
+            
             # Type Email #####
             self.mylog("Type Email", end="")
             el_email.clear()
             el_email.send_keys(self.configuration[PARAM_GRDF_LOGIN])
+            self.mylog(st="OK") 
+
+            re_btn = self.__browser.find_element(
+                By.ID, "grdfButton"
+            )
+            re_btn.click()            
+            time.sleep(3)
+            
+            if self.configuration[PARAM_SCREENSHOT]:
+                self.get_screenshot("02_screenshot_after_user_suivant.png")
+            
+            # Wait for Password #####
+            self.mylog("Waiting for Password", end="")
+
+            ep = EC.presence_of_element_located((By.ID, "input54"))
```

Comment:
For me the IDs were changing on each refresh. It may be more stable to rely on input types or input name
I used this for password
```python
            ep = EC.presence_of_element_located(
                (By.XPATH, r"//input[@name='credentials.passcode']")
            )
```
and this for email
```python
           EC.presence_of_element_located(
                (By.XPATH, r"//input[@name='identifier']")
            )
```